File: 백준/Silver/1021. 회전하는 큐/회전하는 큐.py
import sys
from collections import deque

input = sys.stdin.readline
cnt = 0
n,m = map(int,input().split())
dq = deque(range(1,n+1))

for i in list(map(int,input().split())):
    while True:
        if i == dq[0]: 
            dq.popleft()
            break
        else:
            if dq.index(i) < len(dq)/2:
                while dq[0] != i:
                    cnt += 1
                    dq.rotate(-1)
            else:
                while dq[0] != i:
                    cnt += 1
                    dq.rotate(1)
print(cnt)

File: 백준/Silver/1021. 회전하는 큐/회전하는 큐.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st1 = new StringTokenizer(br.readLine());
        StringTokenizer st2 = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st1.nextToken());
        int m = Integer.parseInt(st1.nextToken());
        int cnt = 0;

        Deque<Integer> dq = new ArrayDeque<>();
        for (int i = 1; i <= n; i++) dq.add(i);

        for (int i = 0; i < m; i++) {
            int x = Integer.parseInt(st2.nextToken());
            while (true) {
                if (x == dq.getFirst()) {
                    dq.removeFirst();
                    break;
                }
                else {
                    if (indexOf(dq, x) < (double) dq.size()/2) {
                        while (dq.getFirst() != x) {
                            cnt ++;
                            dq.addLast(dq.removeFirst());
                        }
                    } else {
                        while (dq.getFirst() != x) {
                            cnt ++;
                            dq.addFirst(dq.removeLast());
                        }
                    }
                }
            }
        }
        System.out.println(cnt);
    }
    private static double indexOf(Deque<Integer> deque, int target) {
        Iterator<Integer> dq = deque.iterator();
        double index = 0;

        while (dq.hasNext()) {
            if (dq.next() == target) {
                return index;
            }
            index++;
        }

        return -1;
    }
}